README.md

```diff
@@ -48,11 +48,45 @@ secret key used to authenticate the session. Inside the handler, we call
 session values in session.Values, which is a `map[interface{}]interface{}`.
 And finally we call `session.Save()` to save the session in the response.
 
-Important Note: If you aren't using gorilla/mux, you need to wrap your handlers
-with
+## Avoiding Memory Leaks
+
+If you are not using gorilla/mux (or are using context), you will
+need to perform additional cleanup or else you will leak memory!
+
+Here is an example that illustrates proper cleanup:
+
+```go
+	import (
+		"context"
+		"net/http"
+
+		gorillacontext "github.com/gorilla/context"
+		"github.com/gorilla/sessions"
+	)
+
+	var store = sessions.NewCookieStore([]byte("something-very-secret"))
+
+	func MyHandler(w http.ResponseWriter, r *http.Request) {
+		// Store some context on the request. This creates
+		// a shallow copy of the request object.
+		r = r.WithContext(context.TODO())
+
+		// Get a session. We're ignoring the error resulted from decoding an
+		// existing session: Get() always returns a session, even if empty.
+		session, _ := store.Get(r, "session-name")
+
+		// Clear the registry at the end of the request lifecycle
+		defer gorillacontext.Clear(r)
```

Comment:
Right, so our code was doing that but we still ran into a memory leak. Specifically, code such as

```
r = r.WithContext(context.TODO())
...
session, _ := store.Get(r, "session-name")
`

being declared within the handler prevents `ClearHandler` from working correctly - the call that ClearHandler makes to clear the context registry ([here](https://github.com/gorilla/context/blob/08b5f424b9271eedf6f9f0ce86cb9396ed337a42/context.go#L140)) still retains a pointer to the original request `r`. But it has been reassigned before reaching `session.Get` and therefore the key to [this map](https://github.com/gorilla/context/blob/08b5f424b9271eedf6f9f0ce86cb9396ed337a42/context.go#L15) contains a different pointer that won't get cleared out by `ClearHandler`.